### PR TITLE
Initial setup for electron socket

### PR DESCRIPTION
**Pull request by laisspportugal**
Description

This PR intends to create an initial setup for electron-socket project. What is included:

- Renaming the project pointing lichtblick projects and references
- Change version on package.json
- Update needed dependencies
- Solve lint issues

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,12 +41,12 @@
     "electron": ">=12"
   },
   "devDependencies": {
-    "@foxglove/eslint-plugin": "0.21.0",
+    "@lichtblick/eslint-plugin": "1.0.1",
     "@foxglove/tsconfig": "^1.1.0",
```
Comment (luluiz) [inline package.json]: Already published
```suggestion
    "@lichtblick/tsconfig": "1.0.0",
```

File: example/tsconfig.json
Comment (luluiz) [inline example/tsconfig.json:1]:
```
  "extends": "@lichtblick/tsconfig/base",
```
ln:2

File: src/preloader/TcpSocketElectron.ts [hunk 2]
```diff
@@ -14,55 +14,95 @@ export class TcpSocketElectron {
...
+          })
+          .catch((err: Error) => {
+            this.#apiResponse(callId, String(err.stack ?? err));
+          });
+      },
+    ],
+    [
+      "close",
+      (callId) => {
+        this.close();
```
Comment (luluiz) [inline src/preloader/TcpSocketElectron.ts]: Here the lint changed the sequence of events.
this.close() -> destroy socket 
this.dispose() -> close connection and destroy socket
this.#apiResponse -> sends a message on the socket and, optionally via args, adds more actions/messages.

The change is closing the connection and then wanting to send a message on a closed connection. Which is not correct.

Initially, I would just say to invert the sequence. However, I saw that it is just changing the return type of this.dispose(), stopping it from being void. For example, change it to string and return "Connection closed", and the existing structure starts working again and lint passes.

I'll make some suggestions to reproduce my idea...

File: src/preloader/TcpSocketElectron.ts [hunk 3]
```diff
@@ -14,55 +14,95 @@ export class TcpSocketElectron {
...
+        this.#apiResponse(callId);
+      },
+    ],
+    [
+      "dispose",
+      (callId) => {
+        this.dispose();
+        this.#apiResponse(callId);
       },
```
Comment (luluiz) [inline src/preloader/TcpSocketElectron.ts:93]: back to the original behaviour, passing `this.dipose()` via params.

File: src/preloader/TcpSocketElectron.ts [hunk 4]
```diff
@@ -115,57 +164,58 @@ export class TcpSocketElectron {
...
+    this.#socket.destroy();
   }
 
   dispose(): void {
-    this._socket.removeAllListeners();
+    this.#socket.removeAllListeners();
     this.close();
-    this._messagePort.close();
+    this.#messagePort.close();
   }
```
Comment (luluiz) [inline src/preloader/TcpSocketElectron.ts]: Change return type to string and returns a message like "Connection disposed"

```suggestion
  dispose(): string {
    this.#socket.removeAllListeners();
    this.close();
    this.#messagePort.close();
    return "Connection disposed";
  }
```

File: src/preloader/TcpSocketElectron.ts (hunk 3 above)
Comment (luluiz) [reply src/preloader/TcpSocketElectron.ts:93]:
```suggestion
        this.#apiResponse(callId, this.dispose());
```

File: src/preloader/TcpSocketElectron.ts [hunk 5]
```diff
@@ -115,57 +164,58 @@ export class TcpSocketElectron {
...
+          this.#emit("connect");
         })
         .on("error", reject);
     });
   }
 
   close(): void {
-    this._socket.destroy();
+    this.#socket.destroy();
   }
```
Comment (luluiz) [inline src/preloader/TcpSocketElectron.ts]:
```suggestion
  close(): string {
    this.#socket.destroy();
    return "Connection closed";
  }
```

File: src/preloader/TcpSocketElectron.ts [hunk 6]
```diff
@@ -14,55 +14,95 @@ export class TcpSocketElectron {
...
+            this.#apiResponse(callId, String(err.stack ?? err));
+          });
+      },
+    ],
+    [
+      "close",
+      (callId) => {
+        this.close();
+        this.#apiResponse(callId);
+      },
```
Comment (luluiz) [inline src/preloader/TcpSocketElectron.ts:86]: original code
```suggestion
        this.#apiResponse(callId, this.close());
```

File: src/preloader/TcpSocketElectron.ts [hunk 7]
```diff
@@ -14,55 +14,95 @@ export class TcpSocketElectron {
...
+          .catch((err: Error) => {
+            this.#apiResponse(callId, String(err.stack ?? err));
+          });
+      },
+    ],
+    [
+      "close",
+      (callId) => {
+        this.close();
+        this.#apiResponse(callId);
```
Comment (luluiz) [inline src/preloader/TcpSocketElectron.ts]: original code
```suggestion
        this.#apiResponse(callId, this.close());
```

**Review by luluiz — CHANGES_REQUESTED** [review]

File: src/preloader/TcpSocketElectron.ts [hunk 8]
```diff
@@ -14,55 +14,95 @@ export class TcpSocketElectron {
...
+      },
+    ],
+    [
+      "dispose",
+      (callId) => {
+        this.dispose();
+        this.#apiResponse(callId);
```
Comment (luluiz) [inline src/preloader/TcpSocketElectron.ts]: back to the original behaviour, passing `this.dipose()` via params.

```suggestion
        this.#apiResponse(callId, this.dispose());
```

File: src/preloader/UdpSocketElectron.ts [hunk 9]
```diff
@@ -141,10 +174,16 @@ export class UdpSocketElectron {
...
-        this._apiResponse(callId);
+        this.#apiResponse(callId);
+      },
+    ],
+    [
+      "dispose",
+      (callId) => {
+        this.dispose();
+        this.#apiResponse(callId);
       },
```
Comment (luluiz) [inline src/preloader/UdpSocketElectron.ts:184]:
```suggestion
        this.#apiResponse(callId, this.dispose());
```

File: src/preloader/UdpSocketElectron.ts [hunk 10]
```diff
@@ -204,66 +255,66 @@ export class UdpSocketElectron {
...
+    this.#socket.disconnect();
   }
 
   dispose(): void {
-    this._socket.removeAllListeners();
+    this.#socket.removeAllListeners();
     void this.close();
-    this._messagePort.close();
+    this.#messagePort.close();
   }
```
Comment (luluiz) [inline src/preloader/UdpSocketElectron.ts]:
```suggestion
  dispose(): string {
    this.#socket.removeAllListeners();
    void this.close();
    this.#messagePort.close();
    return "Connection disposed";
  }
```

File: src/preloader/TcpSocketElectron.ts (hunk 5 above)
Comment (laisspportugal) [reply src/preloader/TcpSocketElectron.ts]: Thanks for the hint, I'm changing to implement it like you suggested.

File: src/preloader/TcpSocketElectron.ts (hunk 4 above)
Comment (laisspportugal) [reply src/preloader/TcpSocketElectron.ts]: Changed, thanks for the suggestion

File: src/preloader/TcpSocketElectron.ts (hunk 7 above)
Comment (laisspportugal) [reply src/preloader/TcpSocketElectron.ts]: Done

File: src/preloader/TcpSocketElectron.ts (hunk 8 above)
Comment (laisspportugal) [reply src/preloader/TcpSocketElectron.ts]: Done

File: src/preloader/mdns.ts [hunk 11]
```diff
@@ -19,7 +19,7 @@ export async function mdns4Request(
   return await new Promise((resolve, reject) => {
     const interfaces = allInterfaces();
     if (interfaces.length === 0) {
-      return reject(new Error("No interfaces to send an mDNS request"));
+      reject(new Error("No interfaces to send an mDNS request"));
     }
```
Comment (luluiz) [inline src/preloader/mdns.ts]: To preserve the original behaviour you need to add return to stop the function.
I've just tested it and just `reject` isn't enough to stop the promise.

```suggestion
      reject(new Error("No interfaces to send an mDNS request"));
      return;
```

File: src/preloader/TcpSocketElectron.ts (hunk 2 above)
Comment (luluiz) [inline src/preloader/TcpSocketElectron.ts:84]: remove this line

File: src/preloader/TcpSocketElectron.ts [hunk 12]
```diff
@@ -14,55 +14,95 @@ export class TcpSocketElectron {
...
+      "close",
+      (callId) => {
+        this.close();
+        this.#apiResponse(callId, this.close());
+      },
+    ],
+    [
+      "dispose",
+      (callId) => {
+        this.dispose();
```
Comment (luluiz) [inline src/preloader/TcpSocketElectron.ts]: remove this line

File: src/preloader/TcpSocketElectron.ts (hunk 2 above)
Comment (luluiz) [inline src/preloader/TcpSocketElectron.ts]: remove this line

File: src/preloader/TcpServerElectron.ts [hunk 13]
```diff
@@ -7,45 +7,72 @@ import { TcpAddress } from "../shared/TcpTypes.js";
...
+          .catch((err: Error) => {
+            this.#apiResponse([callId, String(err.stack ?? err)]);
+          });
+      },
+    ],
+    [
+      "close",
+      (callId) => {
+        this.close();
+        this.#apiResponse([callId, undefined]);
```
Comment (luluiz) [inline src/preloader/TcpServerElectron.ts]: Do the same strategy applied in the last suggestions.
`this.#apiResponse([callId, this.close()]);`

File: src/preloader/TcpServerElectron.ts [hunk 14]
```diff
@@ -7,45 +7,72 @@ import { TcpAddress } from "../shared/TcpTypes.js";
...
+      },
+    ],
+    [
+      "dispose",
+      (callId) => {
+        this.dispose();
+        this.#apiResponse([callId, undefined]);
```
Comment (luluiz) [inline src/preloader/TcpServerElectron.ts]: Do the same strategy applied in the last suggestions.
this.#apiResponse([callId, this.dispose()]);

File: src/preloader/mdns.ts (hunk 11 above)
Comment (laisspportugal) [reply src/preloader/mdns.ts]: Fixed

File: src/preloader/TcpSocketElectron.ts (hunk 2 above)
Comment (laisspportugal) [reply src/preloader/TcpSocketElectron.ts]: Just made the suggested changes. Thanks again for your effort and collaboration.

**Review by luluiz — DISMISSED** [review]

File: src/preloader/UdpSocketElectron.ts [hunk 15]
```diff
@@ -141,10 +174,16 @@ export class UdpSocketElectron {
...
         const multicastInterface = args[2] as string | undefined;
         this.dropSourceSpecificMembership(sourceAddress, groupAddress, multicastInterface);
-        this._apiResponse(callId);
+        this.#apiResponse(callId);
+      },
+    ],
+    [
+      "dispose",
+      (callId) => {
+        this.dispose();
```
Comment (luluiz) [inline src/preloader/UdpSocketElectron.ts]: remove line

**Review by luluiz — APPROVED** [review]
good job

**Review by aneuwald-ctw — APPROVED** [review]
lgtm, thanks for the effort of both of you.

**Pull request merged**